pkg/capr/planner/instructions.go

```diff
@@ -26,41 +27,54 @@ func (p *Planner) generateInstallInstruction(controlPlane *rkev1.RKEControlPlane
 		}
 		switch cattleOS {
 		case capr.WindowsMachineOS:
-			env = append(env, fmt.Sprintf("$env:%s=\"%s\"", arg.Name, arg.Value))
+			env = append(env, capr.FormatWindowsEnvVar(corev1.EnvVar{
+				Name:  arg.Name,
+				Value: arg.Value,
+			}, true))
 		default:
 			env = append(env, fmt.Sprintf("%s=%s", arg.Name, arg.Value))
 		}
 	}
 	switch cattleOS {
 	case capr.WindowsMachineOS:
+		// TODO: Properly format the data dir when adding full support for Windows nodes
 		env = append(env, fmt.Sprintf("$env:%s_DATA_DIR=\"c:%s\"", strings.ToUpper(capr.GetRuntime(controlPlane.Spec.KubernetesVersion)), capr.GetDistroDataDir(controlPlane)))
+		env = append(env, capr.FormatWindowsEnvVar(corev1.EnvVar{
+			Name:  "INSTALL_RKE2_VERSION",
+			Value: controlPlane.Spec.KubernetesVersion,
+		}, true))
```

Comment:
Are you referring to line 41 compared to line 42? If so, thats intentional. I could comment that line out if we want, it is related to the rke2 data dir feature that went out a few releases ago. 

Unfortunately that implementation hit the same formatting issues that all other Windows env vars hit, so it has never worked. The RFC for that data-dir feature did not consider Windows, and there are several other places that need to be updated to have everything work as expected. I'd rather not touch anything related to the feature until we can create a follow-up RFC for windows support, so I did not touch that env var.

Reply:
Yes,  that is what I am referring to. Sorry that I did not make it clear. 

Thanks for the explanation. Just to confirm my understanding: the `env` variable in this function is used in Rancher's Plan (not in the PowerShell script), and each element should follow the format `Name=Var`. However, the `date-dir` environment variable generated on line 41 is incorrectly formatted, but we leave it as is. 

Does it sound right to you?

Reply:
Correct. Since the system-agents applyinator will [execute plans using `cmd.Exec`](https://github.com/rancher/system-agent/blob/7ad21ffbe9273d33d28fbf17249e949fdc6d8610/pkg/applyinator/applyinator.go#L504-L533), which automatically prepends `$env:` to variable names on windows, the `system-agent-installer-rke2` script would receive an environment variable with the name `$env:$env:RKE2_DATA_DIR`. If we were to properly format that variable in this PR, other operations done on WIndows nodes would not respect the new data-dir and would fail to find specific files on the disk.